### PR TITLE
Support optional version and tag

### PR DESCRIPTION
**Pull request by stuartmscott**
State Machines identities can now include an optional version to enable multiple versions of the same binary and tag to distinguish multiple instances of the same binary;
- `package`/`name`@`address`
- `package`/`name`:`tag`@`address`
- `package`/`nameversion`@`address`
- `package`/`nameversion`:`tag`@`address`

Fixes: #11

### CODE REVIEW

**Comment by winksaville** [comment]
My first reaction is that it's overly complex. And makes a name with an "_v" in it potentially ambiguous. I'd suggest defining a way to add version to a tag unambiguously. Maybe with another separator a suggestion might be ";":

  name:tag;version@address

And then to keep the parsing logic simple only one ":", ";" and "@" are allowed in the "connection string".

One other thing, I'd assume an single actor should be able to handle messages defined for a set of "tag/version" pairs. So, it would seem that information needs to be available to the message handler on a message by message basis. Maybe you're handling that, I haven't looked at the code yet :)

File: test/src/semver.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  ASSERT_EQ("alpha", s2.prerelease());
+  ASSERT_EQ("20230111", s2.build());
+}
+
+TEST(SemVerTest, Compare) {
+  ASSERT_TRUE(SemVer("1.0.0") < SemVer("2.0.0"));
+  ASSERT_TRUE(SemVer("1.0.0") < SemVer("1.1.0"));
+  ASSERT_TRUE(SemVer("1.0.0") < SemVer("1.0.1"));
+  ASSERT_TRUE(SemVer("1.0.0+alpha") < SemVer("1.0.0"));
+  // TODO ASSERT_TRUE(SemVer("1.0.0+alpha") < SemVer("1.0.0+alpha.1"));
```
Comment (winksaville) [inline test/src/semver.cpp:82]: I see no test or TODO for "-" \<pre-release\>?

**Review by winksaville — DISMISSED** [review]
This is a huge change and it's tough to do a review so it's just "skimming".

One thing I see is that your `class SemVer` and [semver.org](https://semver.org/) supports 5 fields (major, minor, patch, prerelease, build) but it appears you've documented that a filename there is <name><version>:<tag>@<address> and your examples imply <version> is only (major, minor, patch) is that correct?

I was looking at `resolveVersion` and the `binary /= machine;` statement had me scratching my head. I [looked it up](https://en.cppreference.com/w/cpp/filesystem/path/append) and apparently its a concatenation of two strings with the appropriate path separator, tricky :)

Again, just a quick review.

File: test/src/semver.cpp (hunk 1 above)
Comment (stuartmscott) [reply test/src/semver.cpp:82]: Ooops I mixed up the signs 🤦

**Comment by stuartmscott** [comment]
I think the formatting in your message got messed up ":@" - same thing happened to my description above.

SemVer doesn't include ':' or '@', but there is still more work needed to support the prerelease and build segments.

Yeah the `std::filesystem::path` concatenation is neat, albeit a little unintuitive.

Appreciate the review!

**Pull request merged**